modules/nw-ptp-configuring-linuxptp-services-as-grandmaster-clock.adoc

```diff
@@ -0,0 +1,89 @@
+// Module included in the following assemblies:
+//
+// * networking/using-ptp.adoc
+
+:_content-type: PROCEDURE
+[id="configuring-linuxptp-services-as-grandmaster-clock_{context}"]
+= Configuring linuxptp services as a grandmaster clock
+
+You can configure the `linuxptp` services (`ptp4l`, `phc2sys`, `ts2phc`) as grandmaster clock by creating a `PtpConfig` custom resource (CR) that configures the host NIC.
+
+The `ts2phc` utility allows you to synchronize the system clock with the PTP grandmaster clock so that the node can stream precision clock signal to downstream PTP ordinary clocks and boundary clocks.
+
+[NOTE]
+====
+Use the following example `PtpConfig` CR as the basis to configure `linuxptp` services as the grandmaster clock for your particular hardware and environment.
+This example CR does not configure PTP fast events. To configure PTP fast events, set appropriate values for `ptp4lOpts`, `ptp4lConf`, and `ptpClockThreshold`.
+`ptpClockThreshold` is used only when events are enabled.
+See "Configuring the PTP fast event notifications publisher" for more information.
+====
+
+.Prerequisites
+
+* Install an Intel Westport Channel network interface in the bare-metal cluster host.
+
+* Install the OpenShift CLI (`oc`).
+
+* Log in as a user with `cluster-admin` privileges.
+
+* Install the PTP Operator.
+
+.Procedure
+
+. Create the following `PtpConfig` CR, and then save the YAML in the `grandmaster-clock-ptp-config.yaml` file.
++
+[[ptp-grandmaster-clock]]
+.Recommended PTP grandmaster clock configuration
```

Comment:
Mention what the default Hardware plugin does ( @josephdrichard   can you share the details?)

Reply:
Added this: https://github.com/openshift/openshift-docs/pull/55704/files#diff-aa3c49f9019e7d101022d927d96d8a5008b52798e8352237c4de357a24969013R23-R27

PTAL

Reply:
We updated this so that setting the plugin mechanism (when enableDefaultConfig) is set will just run the hard-coded script to do the required setup for WPC.